src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -208,64 +245,88 @@ fn save_directory(
   store: Arc<Store>,
   posix_fs: Arc<fs::PosixFS>,
   base_dir: Arc<fs::Dir>,
+  include_parents: bool,
 ) -> BoxFuture<(Fingerprint, usize), String> {
   let store_copy = store.clone();
-  Box::new(
-    posix_fs
-      .scandir(&base_dir)
-      .map_err(move |err| {
-        format!("Failed to scandir {:?}: {}", &base_dir, err.description())
-      })
-      .and_then(move |stats| {
-        let mut dirs = Vec::new();
-        let mut files = Vec::new();
-        for stat in stats.into_iter() {
-          match stat {
-            fs::Stat::Dir(dir) => dirs.push(dir),
-            fs::Stat::File(file) => files.push(file),
-            fs::Stat::Link(fs::Link(path)) => {
-              let err: BoxFuture<_, _> =
-                futures::future::err(format!("Don't yet know how to handle symlinks: {:?}", path))
-                  .to_boxed();
-              return err;
-            }
+  let store_copy2 = store.clone();
```

Comment:
Cloning nearby the usage rather than going for C-style "everything at the top" would be clearer I think.

Reply:
Managed to get rid of the clone :)